### DISCUSSION TITLE
Using encoder with SFRC2D.B Gotek pcb in a Yamaha SY77.

### DISCUSSION THREAD

**Discussion by Narfman96** [keirf/flashfloppy#948] [General]
My first post so a short intro. I have collected and repaired many of the synths from the 80's and 90's and have several with floppy drives. I was successful with a Kurzweil K2VX using FlashFloppy. Next I have two Yamaha SY85's that need the treatment...  
I have the Gotek SFRC2D.B slimline emulator installed in my Yamaha SY77 with FF 3.42 firmware. It has an AT32F415RBT7 chip and OLED display I installed. Jumper RA is removed to get out of flash update mode. I have a cable adapter to go from the 26 pin flat cable to the IDC 26 pin ribbon cable. The emulator works great with the synth but I have two issues. The encoder is action perfect using the pcb and connecting to the header where the program enable jumper is located. Is there a connection for the select switch in the encoder? It's not a deal breaker but I can't find any info for this online. 
The next issue I have is there are only 887 image slots shown. When the USB stick is used in another FF 3.42 Gotek SFRKC30.AT4.35 emulator all 1,000 images are shown. Using the Artery programmer I tried versions from 3.30 all the way to 4.6a beta with no changes in the images available. Except 4.6a only shows 859 images available??? It's not a big deal so I deleted images so there are 750 images on the stick all accessible. The SY77 is never going to need 1,000 disk images in the real world. Thank you in advance for any help offered. Fran Bourdeau aka Narfman96

**Comment by liveboxandy** [comment]
You should be able to place a switch between the RA pad and it's ground. This will function as a select switch. 
You should also be able to place a rotary encoder between U3 Pin 50 (Enc1) Pin 51 (Enc2) and ground.

**Comment by Narfman96** [comment]
Wow! Fast answer. Thank you! I plugged the encoder switch in to RA and it works great. I already had header pins on RA, RB, and RC. The 5v power, ground, DT and CLK are already connected to the programming enable header. This emulator is now 100% functional for hardware. The only issue left is the images area appearing to not having enough room for all 1,000 images...

**Comment by keirf** [comment]
FlashFloppy attempts to read and sort the whole folder, but the 415 chip has only 32kB RAM. When out of RAM, the folder listing gets truncated by default. Options:
1. Change the `folder-sort=` option in FF.CFG (https://github.com/keirf/flashfloppy/wiki/FF.CFG-Configuration-File)
2. Assuming you have an OLED display, put your images in subfolders such that you have fewer images per folder.
